packages/flutter_tools/lib/src/build_system/targets/dart_plugin_registrant.dart

```diff
@@ -65,17 +65,14 @@ class DartPluginRegistrantTarget extends Target {
       return true;
     }
     final String? platformName = environment.defines[kTargetPlatform];
-    if (platformName == null) {
-      return true;
-    }
-    final TargetPlatform? targetPlatform = getTargetPlatformForName(platformName);
-    // TODO(egarciad): Support Android and iOS.
-    // https://github.com/flutter/flutter/issues/52267
-    return targetPlatform != TargetPlatform.darwin &&
-           targetPlatform != TargetPlatform.linux_x64 &&
-           targetPlatform != TargetPlatform.linux_arm64 &&
-           targetPlatform != TargetPlatform.windows_x64 &&
-           targetPlatform != TargetPlatform.windows_uwp_x64;
+    final TargetPlatform? targetPlatform = platformName == null ? null
```

Comment:
nit: what about:

```dart
if (platformName == null) {
  return false;
}
```

Reply:
That's what was there before, and has the effect that if a platform isn't passed it silently does nothing. I'd rather we only opt out for platforms we're sure we don't want to do this, since as a failure mode, failing to do registration silently is much worse that generating a registrant that shouldn't do anything.